slider/replay.py

```diff
@@ -576,6 +636,62 @@ def parse(cls, data, *, library=None, client=None, save=False):
             **mod_kwargs,
         )
 
+    @lazyval
+    def hits(self):
+        b = self.beatmap
+        rdata = self.actions
+        scores = {"perfects": [],
+                  "goods": [],
+                  "bads": [],
+                  "misses": [],
+                  "sliderbreaks": [],
+                  }
+        hw = od_to_ms(b.od(easy=self.easy, hard_rock=self.hard_rock))
+        rad = circle_radius(b.cs(easy=self.easy, hard_rock=self.hard_rock))
+        i = 0
+        for obj in b.hit_objects:
+            if self.hard_rock:
```

Comment:
I do think that moving the stuff in the for loop into a separate function would probably be cleaner but i can't see a way to make the pointer work cleanly and it would also get passed a lot of things (like hw, rad etc.) that make it's signature really ugly. I was kinda thinking of the reasoning behind making a separate function being vaguely that someone might be able to use it on it's own.

I think that with a bit of a restructure it could work but I don't have time at the moment (and won't for a while). Is there a todo list i can put it on or something?